projects/plugins/jetpack/extensions/blocks/map/mapkit-utils/index.js

```diff
@@ -54,9 +56,83 @@ function createCalloutElementCallback( currentDoc, callback ) {
 	};
 }
 
+function waitUntilMapkitIsInitialized( currentWindow ) {
+	return new Promise( resolve => {
+		const check = () => {
```

Comment:
Just flagging that this would be checking forever in case `currentWindow.mapkitIsInitializing` was `undefined`. At the same time I understand that the expectation is that the `currentWindow.mapkitIsInitializing` is already initiliazed when the function is called.

Reply:
It only runs when it's defined, but you're right, this shouldn't be the responsibility of the consumer of this function. I've added an extra check that rejects the returned promise.